marketplace-tests/src/test/java/io/syndesis/qe/MarketplaceTest.java

```diff
@@ -0,0 +1,311 @@
+
+        if (OpenShiftUtils.getInstance().getProject(TestConfiguration.openShiftNamespace()) != null) {
+            log.info("Namespace exists, deleting namespace first");
+            OpenShiftUtils.getInstance().deleteProject(TestConfiguration.openShiftNamespace());
+            TestUtils.waitFor(
+                () -> OpenShiftUtils.getInstance().getProject(TestConfiguration.openShiftNamespace()) == null,
+                1, 30,
+                "Namespace was not deleted");
+        }
+
+        log.info("Creating namespace");
+
+        OpenShiftUtils.asRegularUser(() -> OpenShiftUtils.getInstance().createProjectRequest(TestConfiguration.openShiftNamespace()));
+        TestUtils.sleepIgnoreInterrupt(10 * 1000L);
+
+        log.info("Creating pull secret");
+
+        if (TestConfiguration.syndesisPullSecret() != null) {
+            log.info("Creating a pull secret with name " + TestConfiguration.syndesisPullSecretName());
+            OpenShiftUtils.getInstance().secrets().createOrReplaceWithNew()
+                .withNewMetadata()
+                .withName(TestConfiguration.syndesisPullSecretName())
+                .endMetadata()
+                .withData(TestUtils.map(".dockerconfigjson", TestConfiguration.syndesisPullSecret()))
+                .withType("kubernetes.io/dockerconfigjson")
+                .done();
+        }
+
+        log.info("Creating operatorgroup");
+
+        CustomResourceDefinitionContext operatorGroupCrdContext = new CustomResourceDefinitionContext.Builder()
+            .withGroup("operators.coreos.com")
+            .withPlural("operatorgroups")
+            .withScope("Namespaced")
+            .withVersion("v1alpha2")
+            .build();
+
+        String operatorGroupYaml = readResource("openshift/create-operatorgroup.yaml")
+            .replaceAll("OPENSHIFT_PROJECT", TestConfiguration.openShiftNamespace());
+
+        OpenShiftUtils.getInstance().customResource(operatorGroupCrdContext)
+            .createOrReplace(TestConfiguration.openShiftNamespace(),
+                             new ByteArrayInputStream(operatorGroupYaml.getBytes(UTF8)));
+
+
+        log.info("Creating operator subscription");
+
+        String subscriptionYaml = readResource("openshift/create-subscription.yaml")
+            .replaceAll("PACKAGE_NAME", packageName)
+            .replaceAll("OPENSHIFT_PROJECT", TestConfiguration.openShiftNamespace());
+
+        CustomResourceDefinitionContext subscriptionCrdContext = new CustomResourceDefinitionContext.Builder()
+            .withGroup("operators.coreos.com")
+            .withPlural("subscriptions")
+            .withScope("Namespaced")
+            .withVersion("v1alpha1")
+            .build();
+
+        OpenShiftUtils.getInstance().customResource(subscriptionCrdContext)
+            .createOrReplace(TestConfiguration.openShiftNamespace(),
+                new ByteArrayInputStream(subscriptionYaml.getBytes(StandardCharsets.UTF_8)));
```

Comment:
The `RawCustomResourceOperationsImpl` actually has `public Map<String, Object> createOrReplace(String namespace, String objectAsString)` so no need for the inputstream dance like with regular resources.